Examples/Swift/ManageOfflineRegionsExample.swift

```diff
@@ -0,0 +1,197 @@
+
+            //Begin downloading the map for offline use.
+            pack!.resume()
+
+        }
+    }
+
+    // MARK: - MGLOfflinePack notification handlers
+
+    @objc func offlinePackProgressDidChange(notification: NSNotification) {
+
+        /**
+         Get the offline pack this notification is regarding,
+         and the metadata associated with the offline pack.
+         */
+        // and the associated user info for the pack; in this case, `name = My Offline Pack`
+        if let pack = notification.object as? MGLOfflinePack,
+           let userInfo = NSKeyedUnarchiver.unarchiveObject(with: pack.context) as? [String: String] {
+
+            if pack.progress.countOfResourcesCompleted == pack.progress.countOfResourcesExpected {
+
+                let byteCount = ByteCountFormatter.string(fromByteCount: Int64(pack.progress.countOfBytesCompleted), countStyle: ByteCountFormatter.CountStyle.memory)
+
+                print("Offline pack “\(userInfo["name"] ?? "unknown")” completed: \(byteCount), \(pack.progress.countOfResourcesCompleted) resources")
+
+            }
+        }
+        // Reload the table to update the progress percentage for each offline pack.
+        self.tableView.reloadData()
+
+    }
+
+}
+
+fileprivate extension MGLOfflinePackProgress {
+    var percentCompleted: Float {
+        let percentage = Float((countOfResourcesCompleted / countOfResourcesExpected) * 100)
+        return percentage
+    }
+}
+
+extension ManageOfflineRegionsExample_Swift: UITableViewDelegate, UITableViewDataSource {
+
+    // Create the table view which will display the downloaded regions.
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+
+        if let packs = MGLOfflineStorage.shared.packs {
+            return packs.count
+        } else {
+            return 0
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let label = UILabel()
+        if section == 0 {
+            label.backgroundColor = UIColor.white
+            label.textColor = UIColor.black
+            label.textAlignment = .center
+            label.text = "No Offline Packs Saved"
+            return label
+        }
+        return nil
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+
+        let cell = UITableViewCell.init(style: .subtitle, reuseIdentifier: "cell")
+
+        if let packs = MGLOfflineStorage.shared.packs {
+            let pack = packs[indexPath.row]
+
+            cell.textLabel?.text = "Region \(indexPath.row + 1): size: \(pack.progress.countOfBytesCompleted)"
+            cell.detailTextLabel?.text = "Percent completion: \(pack.progress.percentCompleted)%"
+
+        }
+
+        return cell
+
+    }
+
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        guard let packs = MGLOfflineStorage.shared.packs else { return }
```

Comment:
Just adding an extra space below this line:

```suggestion
        guard let packs = MGLOfflineStorage.shared.packs else { return }
        
```